sdk/core/azure-core-http-okhttp/src/main/java/com/azure/core/http/okhttp/OkHttpAsyncHttpClient.java

```diff
@@ -74,7 +79,13 @@
             .map(rb -> {
                 rb.url(request.getUrl());
                 if (request.getHeaders() != null) {
-                    return rb.headers(okhttp3.Headers.of(request.getHeaders().toMap()));
+                    Map<String, String> headers = new HashMap<>();
+                    for (HttpHeader hdr : request.getHeaders()) {
+                        if (hdr.getValue() != null) {
```

Comment:
null or null or empty string?

Reply:
Empty string as header value supported in Netty & OkHttp.